backend/src/federation/index.ts

```diff
@@ -18,37 +13,141 @@ const POLLTIME = 20000
 const SUCCESSTIME = 120000
 const ERRORTIME = 240000
 const ANNOUNCETIME = 30000
-const nodeRand = between(1, 99)
-const nodeURL = `https://test${nodeRand}.org`
-const nodeAPI = {
-  API_1_00: `${nodeURL}/api/1_00/`,
-  API_1_01: `${nodeURL}/api/1_01/`,
-  API_2_00: `${nodeURL}/graphql/2_00/`,
+
+enum ApiVersionType {
+  V1_0 = 'v1_0',
+  V1_1 = 'v1_1',
+  V2_0 = 'v2_0',
+}
+type CommunityApi = {
+  api: string
+  url: string
 }
 
-export const startDHT = async (
-  // connection: Connection,
-  topic: string,
-): Promise<void> => {
+export const startDHT = async (topic: string): Promise<void> => {
   try {
+    let testModeCtrl = 0
+    const testModeData = [
+      `hello here is a new community and i don't know how to communicate with you`,
+      [`string1`, `api`, `url3`],
+      [
+        [`api`, `url`, `wrong`],
+        [`wrong`, `api`, `url`],
+      ],
+      [
+        { wrong: 'wrong property name test', api: 'api1', url: 'url1' },
+        { api: 'api2', url: 'url2', wrong: 'wrong property name test' },
+      ],
+      [
+        { test1: 'api proterty name test', url: 'any url definition as string' },
+        { api: 'some api', test2: 'url property name test' },
+      ],
+      [
+        { api: 1, url: 'api number type test' },
+        { api: 'urltyptest', url: 2 },
+      ],
+      [
+        {
+          api: ApiVersionType.V1_0,
+          url: CONFIG.FEDERATION_COMMUNITY_URL
+            ? (CONFIG.FEDERATION_COMMUNITY_URL.endsWith('/')
+                ? CONFIG.FEDERATION_COMMUNITY_URL
+                : CONFIG.FEDERATION_COMMUNITY_URL + '/') + ApiVersionType.V1_0
+            : 'not configured',
+        },
+        {
+          api: ApiVersionType.V2_0,
+          url: CONFIG.FEDERATION_COMMUNITY_URL
+            ? (CONFIG.FEDERATION_COMMUNITY_URL.endsWith('/')
+                ? CONFIG.FEDERATION_COMMUNITY_URL
+                : CONFIG.FEDERATION_COMMUNITY_URL + '/') + ApiVersionType.V2_0
+            : 'not configured',
+        },
+      ],
+    ]
     const TOPIC = DHT.hash(Buffer.from(topic))
     const keyPair = DHT.keyPair(getSeed())
     logger.info(`keyPairDHT: publicKey=${keyPair.publicKey.toString('hex')}`)
     logger.debug(`keyPairDHT: secretKey=${keyPair.secretKey.toString('hex')}`)
 
+    const ownApiVersions = Object.values(ApiVersionType).map(function (apiEnum) {
+      const comApi: CommunityApi = {
+        api: apiEnum,
+        url: CONFIG.FEDERATION_COMMUNITY_URL
+          ? (CONFIG.FEDERATION_COMMUNITY_URL.endsWith('/')
+              ? CONFIG.FEDERATION_COMMUNITY_URL
+              : CONFIG.FEDERATION_COMMUNITY_URL + '/') + apiEnum
+          : 'not configured',
```

Comment:
this part should be done in the config.
`not configured` is not a good value to be transmitted.